.github/gradle/gradle.gradle

```diff
@@ -1,41 +1,131 @@
+import groovy.json.JsonBuilder
+import groovy.transform.CompileStatic
+
 if (!project.hasProperty('githubCiTesting') && System.env['GITHUB_OUTPUT'] == null) {
     // If the property is not set and we are not running in the CI environment then we will do nothing
     return
 }
 
-static def outputUnitTest(Project project, Task test) {
-    project.getLogger().lifecycle("<< TEST >>${test.path}")
+static Map<String, String> removeCommonPrefix(List<String> list) {
+    if (list.size() == 0) {
+        return new HashMap<String, String>()
+    }
+
+    def result = new HashMap<String, String>()
+
+    if (list.size() == 1) {
+        System.out.println("Only one item in the list: ${list}")
+        def sections = list[0].split("\\.")
+        result.put(list[0], sections.last())
+        return result
+    }
+
+    Collections.sort(list, { String a, String b -> a.length() - b.length() })
+    def max = list[0].length()
+    def prefix = ""
+    for(int index = 0; index < max; index++) {
+        def currentChar = list[0][index]
+        for (String item : list) {
+            if (item[index] != currentChar) {
+                for (final def entry in list) {
+                    result.put(entry, entry.substring(prefix.length()))
+                }
+                return result
+            }
+        }
+
+        prefix += currentChar
+    }
+
+    throw new RuntimeException("Could not remove common prefix: ${list}")
```

Comment:
if I understand this correctly, there are two cases where we might be unable to strip a common prefix:

1. All strings in the list had no common prefix, so `prefix` is empty; or
2. All strings in the list are the exact same, where `prefix` is non-empty.

In all other cases, there is a common prefix among unique strings.

If we follow the suggestion in my other review comment to move the `if` condition outside of the loop, we can gate this `RuntimeException` (which should really be an `IllegalArgumentException` I think) behind an `if (prefix.isEmpty() || list.unique(false).size() == 1)`